retrievalmarket/impl/clientstates/client_fsm.go

```diff
@@ -28,26 +29,41 @@ var ClientEvents = fsm.Events{
 	fsm.Event(rm.ClientEventOpen).
 		From(rm.DealStatusNew).ToNoChange(),
 	fsm.Event(rm.ClientEventPaymentChannelErrored).
-		From(rm.DealStatusAccepted).To(rm.DealStatusFailed).
+		FromMany(rm.DealStatusAccepted, rm.DealStatusPaymentChannelCreating).To(rm.DealStatusFailed).
 		Action(func(deal *rm.ClientDealState, err error) error {
-			deal.Message = xerrors.Errorf("getting payment channel: %w", err).Error()
+			deal.Message = xerrors.Errorf("get or create payment channel: %w", err).Error()
 			return nil
 		}),
-	fsm.Event(rm.ClientEventAllocateLaneErrored).
-		From(rm.DealStatusAccepted).To(rm.DealStatusFailed).
-		Action(func(deal *rm.ClientDealState, err error) error {
-			deal.Message = xerrors.Errorf("allocating payment lane: %w", err).Error()
+	fsm.Event(rm.ClientEventPaymentChannelCreateInitiated).
+		From(rm.DealStatusAccepted).To(rm.DealStatusPaymentChannelCreating).
+		Action(func(deal *rm.ClientDealState, msgCID cid.Cid) error {
+			deal.WaitMsgCID = &msgCID
 			return nil
 		}),
-	fsm.Event(rm.ClientEventPaymentChannelCreated).
-		From(rm.DealStatusAccepted).To(rm.DealStatusPaymentChannelCreated).
+	fsm.Event(rm.ClientEventPaymentChannelAddingFunds).
+		FromMany(rm.DealStatusOngoing, rm.DealStatusPaymentChannelReady).To(rm.DealStatusPaymentChannelAddingFunds),
```

Comment:
The only start state that makes sense here is rm.DealStatusAccepted, it's the the SetupPaymentChannelStart that creates the dispatches this event, which gets called as the state handler for DealStatusAccepted